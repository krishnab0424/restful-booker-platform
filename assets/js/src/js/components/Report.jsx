import React from 'react';
import { Calendar, momentLocalizer} from 'react-big-calendar';
import moment from 'moment/moment';
import AdminBooking from './AdminBooking.jsx';

const fetch = require('node-fetch');

import "react-big-calendar/lib/css/react-big-calendar.css";
import { API } from '../libs/Api';

export default class Report extends React.Component {

  constructor(){
    super();
    
    this.state = {
      report : [],
      currentTimestamp : moment, 
      showBookingForm : false,
      dates : null
    }

    this.addBooking = this.addBooking.bind(this);
    this.closeBooking = this.closeBooking.bind(this);
  }

  componentDidMount(){
    API.getReport(this);
  }

  addBooking(result){
    if(result.slots.length > 1){
      this.setState({showBookingForm : true, dates : result});
    }
  }

  closeBooking(){
    this.setState({showBookingForm : false});
    API.getReport(this);
  }

  render(){
    const localizer = momentLocalizer(this.state.currentTimestamp);

    if(this.state.showBookingForm){
      return <AdminBooking closeBooking={this.closeBooking} dates={this.state.dates} />
    } else {
      return <div>
        <Calendar
          onSelectSlot={this.addBooking}
          selectable
          localizer={localizer}
          defaultView="month"
          popup={true}
          events={this.state.report}
          style={{ height: "75vh" }}
          views={['month']}
        />
      </div>
    }
  }
}