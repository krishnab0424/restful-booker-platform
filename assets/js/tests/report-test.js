import React from 'react';
import Report from '../src/js/components/Report.jsx';
import moment from 'moment/moment';
import nock from 'nock';

test('Multiple reports can be created in the Report component', (done) => {
    nock('http://localhost')
      .get('/report/')
      .reply(200,  {
        currentTimestamp : moment.utc([2019, 0, 1, 1]),
        report : [
          {
            start: "2019-04-01",
            end: "2019-04-03",
            title: "101"
          },{
            start: "2019-04-02",
            end: "2019-04-04",
            title: "102"
          },{
            start: "2019-04-02",
            end: "2019-04-04",
            title: "103"
          }
        ]
      });

    const reportComponent = shallow(
        <Report />
    )

    setTimeout(() => {
      expect(reportComponent).toMatchSnapshot();
      done()
    }, 100);
});