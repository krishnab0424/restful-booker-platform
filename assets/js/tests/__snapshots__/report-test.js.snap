// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Multiple reports can be created in the Report component 1`] = `
<div>
  <ForwardRef
    defaultView="month"
    events={
      Array [
        Object {
          "end": "2019-04-03",
          "start": "2019-04-01",
          "title": "101",
        },
        Object {
          "end": "2019-04-04",
          "start": "2019-04-02",
          "title": "102",
        },
        Object {
          "end": "2019-04-04",
          "start": "2019-04-02",
          "title": "103",
        },
      ]
    }
    localizer={
      DateLocalizer {
        "format": [Function],
        "formats": Object {
          "agendaDateFormat": "ddd MMM DD",
          "agendaHeaderFormat": [Function],
          "agendaTimeFormat": "LT",
          "agendaTimeRangeFormat": [Function],
          "dateFormat": "DD",
          "dayFormat": "DD ddd",
          "dayHeaderFormat": "dddd MMM DD",
          "dayRangeHeaderFormat": [Function],
          "eventTimeRangeEndFormat": [Function],
          "eventTimeRangeFormat": [Function],
          "eventTimeRangeStartFormat": [Function],
          "monthHeaderFormat": "MMMM YYYY",
          "selectRangeFormat": [Function],
          "timeGutterFormat": "LT",
          "weekdayFormat": "ddd",
        },
        "propType": [Function],
        "startOfWeek": [Function],
      }
    }
    onSelectSlot={[Function]}
    popup={true}
    selectable={true}
    style={
      Object {
        "height": "75vh",
      }
    }
    views={
      Array [
        "month",
      ]
    }
  />
</div>
`;
